src/Control/Carrier/Interpret.hs

```diff
@@ -1,5 +1,8 @@
 {-# LANGUAGE FlexibleContexts, FlexibleInstances, FunctionalDependencies, GeneralizedNewtypeDeriving, KindSignatures, RankNTypes, ScopedTypeVariables, TypeApplications, TypeOperators, UndecidableInstances #-}
-
+-- | Provides an 'InterpretC' carrier capable of interpreting an arbitrary effect using
+-- a passed-in higher order function to interpret that effect. The resulting code will be
+-- less efficient than using a carrier monad and 'Carrier' instance directly, but this
+-- module is suitable for prototyping new effects quickly, or when efficiency is not a concern.
```

Comment:
I want to provide an example of how to use `InterpretC` here; I will do that later, as I am tired of looking at docstrings at the moment.